### PR TITLE
 Add an example of decoding transfer events using the built-in decoder.

### PR DESCRIPTION
**Pull request by JasoonS**
I wasn't sure exactly how to approach reviewing your code so decided to try and make an example instead and I think it looks really cool and works very smoothly. I do have a question that I'll put on the code itself.

### CODE REVIEW

File: examples/erc721_events_decoded.go [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+	"github.com/enviodev/hypersync-client-go/decoder"
+	"github.com/enviodev/hypersync-client-go/logger"
+	"github.com/enviodev/hypersync-client-go/options"
+	"github.com/enviodev/hypersync-client-go/types"
+	"github.com/enviodev/hypersync-client-go/utils"
+	"github.com/ethereum/go-ethereum/common"
+	"go.uber.org/zap"
+)
+
+func main() {
```
Comment (JasoonS) [inline examples/erc721_events_decoded.go:23]: I wasn't sure what this error is about... The code runs fine at the very least. Didn't research it.

File: examples/erc721_events_decoded.go [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+
+	logger.L().Info("Fetching USDC transfer logs", zap.Any("start_block", startBlock), zap.Any("end_block", endBlock))
+
+	// Clovers Transfer event signature: Transfer(address,address,uint256)
+	selections := []types.LogSelection{
+		{
+			Address: []common.Address{common.HexToAddress("0xB55C5cAc5014C662fDBF21A2C59Cd45403C482Fd")}, // Clovers contract address
+			Topics: [][]common.Hash{
+				{common.HexToHash("0xddf252ad1be2c89b69c2b068fc378daa952ba7f163c4a11628f55a4df523b3ef")}, // Transfer event signature
+			},
```
Comment (JasoonS) [inline examples/erc721_events_decoded.go:61]: Do I have to create a new selection for every log type that I want to get from a contract?  What if I wanted to get the `ApprovalForAll` event also from this contract?

Comment (0x19) [reply examples/erc721_events_decoded.go:61]: In this specific case yes. We didn't implement loading contracts and abis to figure out available events for contracts nor do we have storage of these information yet in go. This is just a basic very fast example.

File: examples/erc721_events_decoded.go (hunk 1 above)
Comment (0x19) [reply examples/erc721_events_decoded.go:23]: This is related to golang compiler, without build tags you cannot have two main() functions in the same project. Main is used as main access endpoint to applications so it's natural that it fails. Should be or sub folder for this particular example or ignored via build tags + executed via specific build tag.

File: examples/erc721_events_decoded.go (hunk 2 above)
Comment (0x19) [reply examples/erc721_events_decoded.go:61]: For this abi that's provided in the contracts, events needs to be extracted from it and can be passed here to automatically get all of the ERC-721 events.

**Pull request merged**